packages/menu/src/Menu/HighlightReducer/utils/getUpdatedIndex.ts

```diff
@@ -0,0 +1,92 @@
+import { DescendantsList } from '@leafygreen-ui/descendants';
+import { isDefined } from '@leafygreen-ui/lib';
+
+import type { Direction, Index } from '../highlight';
+
+import { isDescendantsSet } from './isDescendantsSet';
+
+/**
+ * Computes the next index given a direction
+ */
+// prettier-ignore
+function getNextIndex(direction: Direction, currentIndex: number, totalItems: number): number;
+// prettier-ignore
+function getNextIndex(direction: Direction, currentIndex: undefined, totalItems: number): undefined;
+// prettier-ignore
+function getNextIndex(direction: Direction, currentIndex: Index, totalItems: number): Index {
+  if (!isDefined(currentIndex)) return currentIndex;
+
+  switch (direction) {
+    case 'next':
+      return (currentIndex + 1) % totalItems;
+    case 'prev':
+      return (currentIndex - 1 + totalItems) % totalItems;
+    case 'last':
+      return totalItems - 1;
+    case 'first':
+    default:
+      return 0;
+  }
+}
+
+/**
+ * Finds the index of the subsequent `enabled` descendant element
+ */
+function getNextEnabledIndex(
+  direction: Direction,
+  currentIndex: number,
+  descendants: DescendantsList<HTMLElement>,
+): Index {
```

Comment:
I'll leave it to you to determine what makes sense to test but I think this `getNextEnabledIndex` util would be helpful to have tested

do you think it would be helpful to have these `HightlightReducer/utils` files live in the descendants package? (not asking you to do that now but wondering so I can be on the same page if that is something worth consolidating for future refactors)

Reply:
I'm not sure if that should live in `descendants`, since each use case will be pretty tailored to the component. Maybe some more abstract utilities though? Not sure

Reply:
yeah it's a bit hard to say without getting more context of each use case. Maybe a case of implement per component and connect the dots once a pattern is more clear